packages/grafana-ui/src/components/Table/TableNG/types.ts

```diff
@@ -40,7 +44,42 @@ export type FilterType = {
   };
 };
 
-export type TableRow = Record<string, unknown>;
+/* ----------------------------- Table specific types ----------------------------- */
+export interface TableSummaryRow {
+  [columnName: string]: string | number | undefined;
+}
+
+export interface TableColumn extends Column<TableRow, TableSummaryRow> {
+  key: string; // Unique identifier used by DataGrid
+  name: string; // Display name in header
+  field: Field; // Grafana field data/config
+  width?: number | string; // Column width
+  minWidth?: number; // Min width constraint
+  cellClass?: string; // CSS styling
+}
+
+// Possible values for table cells based on field types
+export type TableCellValue =
+  | string // FieldType.string, FieldType.enum
+  | number // FieldType.number
+  | boolean // FieldType.boolean
+  | Date // FieldType.time
+  | DataFrame // For nested data
+  | DataFrame[] // For nested frames
+  | undefined; // For undefined values
+
+export interface TableRow {
+  // Required metadata properties
+  __depth: number;
+  __index: number;
+
+  // Nested table properties
+  data?: DataFrame;
+  'Nested frames'?: DataFrame[];
```

Comment:
:see_no_evil: oh dear lord, we're gonna need to clean this up later. hard to understand why this ever shipped like this.

Reply:
It needs at least 3 underscores.